coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -1394,6 +1399,7 @@ func (s *server) CompleteJob(ctx context.Context, completed *proto.CompletedJob)
 				// This is for deleting a workspace!
 				return nil
 			}
+			s.notifyWorkspaceDeleted(ctx, workspace, workspaceBuild)
```

Comment:
You're calling this from _inside_ a database transaction, and calling out to code that will send the notification outside the transaction.  Couple problems with this:

1. If the transaction gets rolled back, you'll send a notification saying the workspace was deleted when it wasn't.
2. The transaction holds a DB connection, and trying to notify outside the transaction will also try to acquire a DB connection. If we exhaust the connection pool, we can deadlock.